### PR TITLE
Add some panic reasons and fix warnings

### PR DESCRIPTION
**Pull request by rukai**
Without these panic reasons I had to inspect the image-gif source code to see why the panic was happening.

### CODE REVIEW

**Comment by rukai** [comment]
Increased the minimum version of rust to 0.11, I can go higher if there is a minimum you are happy with such that future PRs don't need to bump it any time soon.

**Comment by rukai** [comment]
Bump

**Comment by martinlindhe** [comment]
Citing @HeroicKatora from elsewhere:

> Every version bump below 1.24.1 should be insignificant, which is[ the oldest version in Debian stable](https://packages.debian.org/search?keywords=rust&searchon=names&suite=stable&section=all) and the image crate itself.

**Comment by rukai** [comment]
Ok, changed the CI rust version to 1.24.1

**Comment by rukai** [comment]
I have also fixed the CI link in the readme.

I would like to also change the documentation links to use docs.rs instead of the now nonexistent docs at piston.rs. But that will need to be done in a follow up PR as currently there is no docs at docs.rs which should be fixed by releasing a new version.

**Comment by HeroicKatora** [comment]
Sure. No need to overload this PR.

File: src/common.rs [hunk 1]
```diff
@@ -127,12 +127,12 @@ impl<'a> Default for Frame<'a> {
 }
 
 impl Frame<'static> {
-    
     /// Creates a frame from pixels in RGBA format.
+    /// Panics when the length of pixels does not equal width * height * 4.
```
Comment (HeroicKatora) [inline src/common.rs]: E.g.

```markdown
    /// Creates a frame from pixels in RGBA format.
    ///
    /// # Panics
    /// The length of pixel data must equal `width * height * 4`.
```

Comment (HeroicKatora) [reply src/common.rs]: Should be `if` not `when`.

**Comment by rukai** [comment]
Done

**Comment by HeroicKatora** [comment]
@rukai Since this is your first contribution, you are invited to the collaborators team with write access to ease further contributions. See [the contribution policy](https://github.com/image-rs/organization/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/image-rs/organization/blob/master/CODE_OF_CONDUCT.md). I can send you an invite if you want.

**Comment by HeroicKatora** [comment]
And thanks!

**Comment by rukai** [comment]
Yes that would be appreciated.

**Pull request merged**